### PR TITLE
feat: promote Teams agent

### PR DESCRIPTION
**Pull request by yuqizhou77**
**[All strings below will be updated later based on what provided by supkasar]**
- sample gallery
image

- new project
image

- Treeview (in Development section)
image

- Get started
image
image

After clicking the button
- If installed and logged in:
chatinstalled

if triggered from Treeview, "@teams" will be put in the textbox, otherwise, the text is "@teams /create"
- not logged in:
notsignedin
- not installed
notinstalled

**UT**
image

core:
image

[User Story 27246583](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/27246583): Promote Teams Agent

### CODE REVIEW

**Comment by yuqizhou77** [comment]
> I approved for telemetry.

updated to send telemetry with error message defined by us. For now, still print the original error from VSCode to the user for easier mitigation.

**Review by tecton — APPROVED** [review]

**Comment by tecton** [comment]
The GitHub Copilot Chat dependency is removed in this PR.

File: packages/fx-core/src/question/create.ts [hunk 1]
```diff
@@ -91,6 +91,7 @@ export class ProjectTypeOptions {
         "core.TabOption.label"
       )}`,
       detail: getLocalizedString("core.createProjectQuestion.projectType.tab.detail"),
+      groupName: "Create",
```
Comment (tecton) [inline packages/fx-core/src/question/create.ts:94]: Does this name need localization?

File: packages/vscode-extension/src/copilotChatHandlers.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      isPartialQuery: true,
+    };
+    await vscode.commands.executeCommand("workbench.panel.chat.view.copilot.focus");
+    await vscode.commands.executeCommand("workbench.action.chat.open", options);
+    return ok(null);
+  } catch (e) {
+    const error = new SystemError(
+      eventName,
+      "InstallCopilotError",
+      localize("teamstoolkit.handlers.openCopilotChatErro"),
```
Comment (tecton) [inline packages/vscode-extension/src/copilotChatHandlers.ts:42]: typo: Erro->Error

Comment (yuqizhou77) [reply packages/vscode-extension/src/copilotChatHandlers.ts]: Thanks, updated

File: packages/fx-core/src/question/create.ts (hunk 1 above)
Comment (yuqizhou77) [reply packages/fx-core/src/question/create.ts]: updated

**Review by Alive-Fish — APPROVED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -301,6 +301,10 @@
   "core.createProjectQuestion.projectType.customCopilot.label": "Custom Copilot",
   "core.createProjectQuestion.projectType.customCopilot.title": "App Features Using Teams AI Library",
   "core.createProjectQuestion.projectType.customCopilot.placeholder": "Select an option",
+  "core.createProjectQuestion.projectType.copilotHelp.label": "Don't know how to start? Use Github Copilot Chat",
+  "core.createProjectQuestion.projectType.copilotHelp.detail": "Chat with Github Copilot to get step by step instructions for Teams app development",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:305]: Proposed: Chat with GitHub Copilot and get step-by-step instructions to develop your Teams app

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -301,6 +301,10 @@
...
+  "core.createProjectQuestion.projectType.copilotGroup.title": "Copilot",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:306]: Proposed: Use Copilot

File: packages/vscode-extension/package.nls.json [hunk 5]
```diff
@@ -134,6 +134,8 @@
     "teamstoolkit.commandsTreeViewProvider.previewDescription": "Debug and preview your Teams app",
     "teamstoolkit.commandsTreeViewProvider.previewTitle": "Preview Your Teams App (F5)",
     "_teamstoolkit.commandsTreeViewProvider.previewTitle.comment": "'F5' is a shortcut key, no need to translate it.",
+    "teamstoolkit.commandsTreeViewProvider.getCopilotHelpTitle": "Get help from Github Copilot",
+    "teamstoolkit.commandsTreeViewProvider.getCopilotHelpDescription": "Chat with Github Copilot to find out what you can do for your current status.",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:138]: Proposed: Chat with GitHub Copilot to know what you can do with your Teams app.

File: packages/vscode-extension/package.nls.json [hunk 6]
```diff
@@ -256,6 +258,13 @@
     "teamstoolkit.localDebug.failedCheckers": "Unable to check: [%s].",
     "teamstoolkit.handlers.askInstallOfficeAddinDependency": "Install dependencies for Office Add-in?",
     "teamstoolkit.handlers.installOfficeAddinDependencyCancelled": "Dependency installation is canceled, but you can install dependencies manually by clicking the 'Development - Check and Install Dependencies' button on the left side.",
+    "teamstoolkit.handlers.askInstallCopilot": "To start with Github Copilot, we recommend you to install the extenstion first.",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:261]: Proposed: To use Github Copilot, install its extension first.

File: packages/vscode-extension/package.nls.json [hunk 7]
```diff
@@ -256,6 +258,13 @@
...
+    "teamstoolkit.handlers.askInstallCopilot.install": "Install",
+    "teamstoolkit.handlers.askInstallCopilot.cancel": "Cancel",
+    "teamstoolkit.handlers.installExtension.output": "You need to install %s first.",
+    "teamstoolkit.handlers.installCopilotError": "Cannot install Github Copilot Chat. Please install it from extension marketplace and try again.",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:265]: Proposed: Unable to install GitHub Copilot Chat. Install it from the extension marketplace and try again.

File: packages/vscode-extension/package.nls.json [hunk 8]
```diff
@@ -256,6 +258,13 @@
...
+    "teamstoolkit.handlers.chatTeamsAgentError": "Cannot focus Github Copilot Chat automatically. Please open Github Copilot Chat and start with \"@teams\"",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:266]: Proposed: Unable to automatically focus GitHub Copilot Chat. Open Github Copilot Chat and start with \"@teams\"

File: packages/vscode-extension/package.nls.json [hunk 9]
```diff
@@ -256,6 +258,13 @@
...
+    "teamstoolkit.handlers.verifyCopilotExtensionError": "Failed to verify Github Copilot Chat. Please install it manually and try again.",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:267]: Proposed: Unable to verify Github Copilot Chat. Install it manually and try again.

File: packages/vscode-extension/package.nls.json [hunk 10]
```diff
@@ -422,9 +431,12 @@
...
+    "teamstoolkit.walkthroughs.steps.teamsToolkitBuildAppWithChat.description": "Start building your first app with [Teams](https://aka.ms/teamsfx-capabilities-overview) or [Outlook add-in](https://aka.ms/teamsfx-outlook-add-in-capabilities) capabilities. Create it from scratch or explore our samples for real-world examples and code structures.\nYou can also start your Teams extension experience with the Github Copilot.\n[Create a New App](command:fx-extension.createFromWalkthrough?%5B%22WalkThrough%22%5D)\n[View Samples](command:fx-extension.openSamples?%5B%22WalkThrough%22%5D)\n [Start with Github Copilot](command:fx-extension.invokeChat?%5B%22WalkThrough%22%5D)\n__Tip: You need to have a Github subscription to use Github Copilot.__",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:439]: Change "You can also start your Teams extension experience with the Github Copilot." to "Enhance your Teams extension experience with GitHub Copilot."

**Review by supkasar — CHANGES_REQUESTED** [review]

File: packages/vscode-extension/package.nls.json (hunk 10 above)
Comment (yuqizhou77) [reply packages/vscode-extension/package.nls.json]: Updated all strings. @supkasar could you help sign off?

File: packages/vscode-extension/package.nls.json [hunk 11]
```diff
@@ -256,6 +258,13 @@
...
     "teamstoolkit.handlers.askInstallOfficeAddinDependency": "Install dependencies for Office Add-in?",
     "teamstoolkit.handlers.installOfficeAddinDependencyCancelled": "Dependency installation is canceled, but you can install dependencies manually by clicking the 'Development - Check and Install Dependencies' button on the left side.",
+    "teamstoolkit.handlers.askInstallCopilot": "To use Github Copilot, install its extension first.",
+    "teamstoolkit.handlers.askInstallCopilot.install": "Install",
+    "teamstoolkit.handlers.askInstallCopilot.cancel": "Cancel",
+    "teamstoolkit.handlers.installExtension.output": "You need to install %s following \"%s\" first.",
+    "teamstoolkit.handlers.installCopilotError": "Unable to install Github Copilot Chat. Install it following %s and try again.",
+    "teamstoolkit.handlers.chatTeamsAgentError": "Unable to automatically focus Github Copilot Chat. Open Github Copilot Chat and start with \"%s\"",
+    "teamstoolkit.handlers.verifyCopilotExtensionError": "Unable to verify Github Copilot Chat. Install it manually following %s and try again.",
     "teamstoolkit.localDebug.learnMore": "Learn More",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:268]: Proposed: Get more info
Logic: As per our style guide, we don't use 'Learn more'.

File: packages/vscode-extension/package.nls.json [hunk 12]
```diff
@@ -256,6 +258,13 @@
...
     "teamstoolkit.localDebug.m365TenantHintMessage": "After enrolling your developer tenant in Office 365 Target Release, enrollment may come into effect in couple of days. Click 'Learn More' button for details on setting up dev environment to extend Teams apps across Microsoft 365.",
```
Comment (supkasar) [inline packages/vscode-extension/package.nls.json:269]: Click 'Get more info' button for....

**Review by supkasar — APPROVED** [review]

**Pull request merged**